<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="robo9">
  <link name="dummy_link" />

  <joint name="dummy_link_to_base_link" type="fixed">
    <parent link="dummy_link"/>
    <child link="base_link"/>
  </joint>
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="-0.000234166671283786 0 0.341965241299168"
        rpy="0 0 0" />
      <mass
        value="291.564682437976" />
      <inertia
        ixx="20.0527856871514"
        ixy="-7.50635775529426E-06"
        ixz="0.0200068652527846"
        iyy="20.1739356871347"
        iyz="-2.14957033006662E-15"
        izz="21.8510802863288" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="left_wheel">
    <inertial>
      <origin
        xyz="0 -5.55111512312578E-17 -0.896942402108544"
        rpy="0 0 0" />
      <mass
        value="2.42815284027599" />
      <inertia
        ixx="0.00682217388768063"
        ixy="-1.15245136693123E-18"
        ixz="6.54305795555698E-19"
        iyy="0.00682217388768063"
        iyz="1.50075961333382E-19"
        izz="0.0107400873808283" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/left_wheel.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/left_wheel.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="left_wheel_joint"
    type="continuous">
    <origin
      xyz="0 0.482600000000004 0.1016"
      rpy="-1.5707963267949 0 0" />
    <parent
      link="base_link" />
    <child
      link="left_wheel" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="right_wheel">
    <inertial>
      <origin
        xyz="-2.22044604925031E-16 0 -0.896942402108544"
        rpy="0 0 0" />
      <mass
        value="2.42815284027599" />
      <inertia
        ixx="0.00682217388768063"
        ixy="-1.88032045455964E-18"
        ixz="-3.31900532483183E-20"
        iyy="0.00682217388768064"
        iyz="-9.58168820167435E-20"
        izz="0.0107400873808283" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/right_wheel.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/right_wheel.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="right_wheel_joint"
    type="continuous">
    <origin
      xyz="0 -0.482600000000004 0.1016"
      rpy="1.5707963267949 0 0" />
    <parent
      link="base_link" />
    <child
      link="right_wheel" />
    <axis
      xyz="0 0 -1" />
  </joint>
  <link
    name="base_to_first_link">
    <inertial>
      <origin
        xyz="-5.55111512312578E-17 -1.11022302462516E-16 -0.1524"
        rpy="0 0 0" />
      <mass
        value="2.47111103406268" />
      <inertia
        ixx="0.0207254059315664"
        ixy="-1.54905385393866E-19"
        ixz="-1.83677921854158E-34"
        iyy="0.0207254059315664"
        iyz="-6.88468379528296E-20"
        izz="0.00318852398947175" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/base_to_first_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/base_to_first_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="base_to_first_link_joint"
    type="fixed">
    <origin
      xyz="0 0 0.7112"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="base_to_first_link" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="first_link">
    <inertial>
      <origin
        xyz="0.00887202489217992 0.166951784287116 -1.02122346670472"
        rpy="0 0 0" />
      <mass
        value="3.1834408940209" />
      <inertia
        ixx="0.0478820519866207"
        ixy="-0.00218702423761148"
        ixz="-1.49937532359432E-05"
        iyy="0.00684333918252238"
        iyz="-0.000282149101960674"
        izz="0.0528634241141966" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/first_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/first_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="first_link_joint"
    type="continuous">
    <origin
      xyz="0 0 1.0236880255875"
      rpy="0 0 0" />
    <parent
      link="base_to_first_link" />
    <child
      link="first_link" />
    <axis
      xyz="0 0 1" />
  </joint>
        <transmission name="trans_first_link">
     <type>transmission_interface/SimpleTransmission</type>
     <joint name="first_link_joint">
       <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
     </joint>
     <actuator name="first_link_motor">
       <mechanicalReduction>10</mechanicalReduction>
       <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>
  <link
    name="second_link">
    <inertial>
      <origin
        xyz="0.00255294857603322 0.0480408165150327 0.00925479847334798"
        rpy="0 0 0" />
      <mass
        value="1.7347084961237" />
      <inertia
        ixx="0.00731696518042928"
        ixy="-0.000136416141636608"
        ixz="9.2876719216566E-06"
        iyy="0.00475716599659306"
        iyz="0.000174773337320167"
        izz="0.0101503487353134" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/second_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/second_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="second_link_joint"
    type="fixed">
    <origin
      xyz="0.0161746276486489 0.304370533101395 -0.947488025587499"
      rpy="0 0 0" />
    <parent
      link="first_link" />
    <child
      link="second_link" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="end_effector_link">
    <inertial>
      <origin
        xyz="-8.11305050484323E-05 -0.00702550005762381 0.0943050421661527"
        rpy="0 0 0" />
      <mass
        value="0.287447503704032" />
      <inertia
        ixx="0.00103661826081033"
        ixy="3.87126226374111E-10"
        ixz="1.97620720820864E-06"
        iyy="0.0014160265512004"
        iyz="2.00867587917765E-05"
        izz="0.000387513820386799" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/end_effector_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/end_effector_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="end_effector_joint"
    type="prismatic">
    <origin
      xyz="-0.0013479 -0.025364 0.0254"
      rpy="1.5708 0 3.0885" />
    <parent
      link="second_link" />
    <child
      link="end_effector_link" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0"
      upper="0.254"
      effort="100"
      velocity="100" />
  </joint>
          <transmission name="trans_end_effector_link">
     <type>transmission_interface/SimpleTransmission</type>
     <joint name="end_effector_joint">
       <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
     </joint>
     <actuator name="end_effector_motor">
       <mechanicalReduction>10</mechanicalReduction>
       <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>
  <link
    name="tray_diffusor">
    <inertial>
      <origin
        xyz="-0.0259895460961588 -0.00224101216839243 0.00327376150152037"
        rpy="0 0 0" />
      <mass
        value="0.620637961492778" />
      <inertia
        ixx="2.4803545805323E-07"
        ixy="-2.30126990196352E-07"
        ixz="-1.6086888268435E-08"
        iyy="5.52745746563034E-06"
        iyz="-1.35859441652098E-09"
        izz="5.73338557318217E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/tray_diffusor.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://robo9/meshes/tray_diffusor.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="tray_diffusor_joint"
    type="revolute">
    <origin
      xyz="-0.050544 -0.00889 0.17146"
      rpy="1.5708 -0.053091 3.1416" />
    <parent
      link="end_effector_link" />
    <child
      link="tray_diffusor" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="3.14"
      upper="6.28"
      effort="100"
      velocity="100" />
  </joint>
        <transmission name="trans_tray_diffusor_link">
     <type>transmission_interface/SimpleTransmission</type>
     <joint name="tray_diffusor_joint">
       <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
     </joint>
     <actuator name="tray_diffusor_motor">
       <mechanicalReduction>10</mechanicalReduction>
       <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </actuator>
  </transmission>
</robot>